demisto_sdk/commands/generate_outputs/generate_descriptions/generate_descriptions.py

```diff
@@ -0,0 +1,289 @@
+                 "\ncontextPath: IDID.Endpoint.host_orchestration_id\ndescriptionMessage: The orchestration object identifier for the Endpoint's host." \
+                 "\ncontextPath: BrublaBRO.Endpoint.is_on\ndescriptionMessage:  Whether the Endpoint is on or not." \
+                 "\ncontextPath: WellDone.Endpoint.metadata.InventoryAPI.report_source\ndescriptionMessage:  The Endpoint's InvestoryAPI report source." \
+                 "\ncontextPath: SomethingElse.Endpoint.bios_uuid\ndescriptionMessage:"
+LOADING_DONE = False
+INPUT_PREFIX = '\r>> Generated output: '
+DEBUG_PROMPT = False
+
+
+def pinput(prompt, prefill=''):
+    import readline
+    readline.set_startup_hook(lambda: readline.insert_text(prefill))
+    try:
+        return input(prompt)
+    finally:
+        readline.set_startup_hook()
+
+
+def clear_prompt():
+    global CURRENT_PROMPT
+    if len(CURRENT_PROMPT) >= 2000 * 4:
+        y = input("Can't teach model anymore (prompt too big), clear?")
+        if y == 'y':
+            print("Truncating prompt:")
+            print(CURRENT_PROMPT)
+            CURRENT_PROMPT = ""
+            print('Teaching from blank now')
+
+
+def correct_prompt(ctx, correct_desc):
+    global CURRENT_PROMPT
+    clear_prompt()
+    CURRENT_PROMPT += f"contextPath: {ctx}\ndescriptionMessage: {correct_desc}\n"
+
+
+def get_current_prompt():
+    global CURRENT_PROMPT
+    return CURRENT_PROMPT
+
+
+def animate():
+    ''' Show a nice animation spinner '''
+    global LOADING_DONE
+    for c in itertools.cycle(['|', '/', '-', '\\']):
+        if LOADING_DONE:
+            break
+        print(INPUT_PREFIX + c, end="", flush=True)
+        time.sleep(0.1)
+
+
+def generate_desc_with_spinner(command_ctx, insecure, output, verbose):
+    global LOADING_DONE
+    LOADING_DONE = False
+    t = threading.Thread(target=animate)
+    t.start()
+    output = generate_desc(command_ctx, verbose=verbose,
+                           prob_check=True,
+                           insecure=insecure)
+    LOADING_DONE = True
+    return output
+
+
+def generate_desc(input_ctx, verbose=False,
+                  prob_check=False, insecure=False):
+    logger = logging.getLogger()
+    logger.setLevel(logging.ERROR)
+
+    prompt = get_current_prompt()
+    prompt += f"contextPath: {input_ctx}\ndescriptionMessage:"
+
+    if verbose:
+        print("=" * 25)
+        print("PROMPT: ")
+        print(prompt)
+        print("=" * 25)
+        print("Using GPT-3...")
+
+    if insecure:
+        return ai21_api_request(prompt, {"prob_check": prob_check,
+                                         "insecure": insecure})
+    return ai21_api_request(prompt, {"prob_check": prob_check})
+
+
+def ai21_api_request(prompt, options={}):
+    res = requests.post(
+        "https://api.ai21.com/studio/v1/j1-large/complete",
+        headers={"Authorization": f"Bearer {os.environ.get('AI21_KEY')}"},
```

Comment:
How does someone get the AI21 key? I guess you will document it later as you said there are missing documents.
Anyway, should print an error/something if key is missing on how to generate it/point to README on how to get it

Reply:
Added docs and fixed.